src/slurm_plugin/fleet_manager.py

```diff
@@ -0,0 +1,386 @@
+            all_or_nothing,
+        )
+
+    def _evaluate_launch_params(self, count, launch_overrides):
+        """Evaluate parameters to be passed to run_instances call."""
+        launch_params = {
+            # Set MinCount to "count" to make the run_instances call fail if entire count cannot be satisfied
+            "MinCount": 1 if not self._all_or_nothing else count,
+            "MaxCount": count,
+            # LaunchTemplate is different for every compute resources in every queue
+            "LaunchTemplate": {
+                "LaunchTemplateName": f"{self._cluster_name}-{self._queue}-{self._compute_resource}",
+                "Version": "$Latest",
+            },
+        }
+
+        launch_params.update(launch_overrides)
+        if launch_overrides:
+            logger.info("Found RunInstances parameters override. Launching instances with: %s", launch_params)
+        return launch_params
+
+    def _launch_instances(self, launch_params):
+        """Launch a batch of ec2 instances."""
+        try:
+            return run_instances(self._region, self._boto3_config, launch_params)
+        except ClientError as e:
+            logger.error("Failed RunInstances request: %s", e.response.get("ResponseMetadata").get("RequestId"))
+            raise e
+
+
+class Ec2CreateFleetManager(FleetManager):
+    """Manager to create EC2 instances fleet using create_fleet API."""
+
+    def __init__(
+        self,
+        cluster_name,
+        region,
+        boto3_config,
+        queue_config,
+        compute_resource_config,
+        all_or_nothing,
+    ):
+        super().__init__(
+            cluster_name,
+            region,
+            boto3_config,
+            queue_config,
+            compute_resource_config,
+            all_or_nothing,
+        )
+
+    def _evaluate_launch_params(self, count, launch_overrides):
+        """Evaluate parameters to be passed to create_fleet call."""
+        overrides = []
+        for instance_type in self._compute_resource_config.get("InstanceTypeList", []):
+            overrides.append({"InstanceType": instance_type["InstanceType"]})
+
+        launch_params = {
+            "LaunchTemplateConfigs": [
+                {
+                    "LaunchTemplateSpecification": {
+                        # LaunchTemplate is different for every compute resources in every queue
+                        "LaunchTemplateName": f"{self._cluster_name}-{self._queue}-{self._compute_resource}",
+                        "Version": "$Latest",
+                    },
+                    "Overrides": overrides,
+                }
+            ],
+            "TargetCapacitySpecification": {
+                "TotalTargetCapacity": count,
+                "DefaultTargetCapacityType": "on-demand"
+                if self._queue_config["CapacityType"] == "ONDEMAND"
+                else "spot",
+            },
+            "Type": "instant",
+            "SpotOptions": {
```

Comment:
can be this set also when launching only on-demand?

Reply:
yes you can fill both of them and then select the one to use with `DefaultTargetCapacityType`